tests/qnn/test_cost.py

```diff
@@ -0,0 +1,44 @@
+import numpy as np
+import pennylane as qml
+
+
+np.random.seed(0)
+
+
+def rx_ansatz(phis, **kwargs):
+    for w, phi in enumerate(phis):
+        qml.RX(phi, wires=w)
+
+
+def layer_ansatz(weights, x=None, **kwargs):
+    qml.templates.AngleEmbedding(x, wires=[0, 1, 2])
+    qml.templates.StronglyEntanglingLayers(weights, wires=[0, 1, 2])
+
+
+class TestMSECost:
+    def test_layer_circuit(self):
+        num_qubits = 3
+
+        dev = qml.device("default.qubit", wires=num_qubits)
+
+        observables = [qml.PauliZ(0), qml.PauliX(0), qml.PauliZ(1) @ qml.PauliZ(2)]
+
+        cost = qml.qnn.MSECost(layer_ansatz, observables, dev)
+        weights = np.random.rand(num_qubits, 3, 3)
```

Comment:
We could make this be `np.ones((num_qubits, 3, 3))` to avoid having to use `np.random.rand` + setting the random seed.